packages/flutter_tools/lib/src/flutter_manifest/parse_result.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+sealed class ParseResult<T> {
```

Comment:
Can this be a more generic result type? I'm thinking:

```
sealed class Result<V, E> {}

final class ValueResult<V, E> implements Result<V, E> {
  ValueResult(this.value);
  
  final V value;
}

final class ErrorResult<V, E> implements Result<V, E> {
  ErrorResult(this.error);
  
  E error;
}
```

Reply:
because we don't constrain the `E` type parameter, it could still be a list of exceptions if we want

Reply:
Is this immediately useful? I'm thinking "YAGNI" here.

I could see this being useful if we decide that we want a similar `Result`-like abstraction in other places in the tool, at which point this could then be refactored

Reply:
why is this use case special, requiring this pattern where we wouldn't need it elsewhere in the tool?

Reply:
It's unique by existence—it's the only currently-known use case for such an abstraction (hence the code being located under the new `flutter_manifest` directory, and not somewhere higher). Why write a more complicated abstraction when it's not yet needed?